In [10]:
!pip install dtale

     --------------------------------------- 14.2/14.2 MB 10.2 MB/s eta 0:00:00
     ---------------------------------------- 45.9/45.9 kB ? eta 0:00:00
     ---------------------------------------- 96.5/96.5 kB ? eta 0:00:00
     ------------------------------------- 219.7/219.7 kB 13.1 MB/s eta 0:00:00
     ---------------------------------------- 10.4/10.4 MB 9.6 MB/s eta 0:00:00
     ------------------------------------- 642.7/642.7 kB 10.2 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
     ---------------------------------------- 65.9/65.9 MB 5.0 MB/s eta 0:00:00
     ---------------------------------------- 62.3/62.3 kB 3.3 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
     -------------------------------------- 383.3/383.3 kB 7.9 MB/s eta 0:00:00
  Created wheel for dash-colorscales: filename=dash_colorscales-0.0.4-py3-none-any.whl siz

In [11]:
import pandas as pd

In [12]:
df = pd.read_csv("Rohit-Sharma-All-International-Cricket-Centuries.csv")

In [13]:
df.head()

No.  Score       Against  Position  Innings  \
0    1    114      Zimbabwe         4        1   
1    2  101*      Sri Lanka         4        2   
2    3  141*      Australia         1        2   
3    4    209     Australia         1        1   
4    5    177   West Indies         6        2   

                               Venue   Ground       Date Result  
0       Queens Sports Club, Bulawayo     Away  28-May-10   Lost  
1       Queens Sports Club, Bulawayo  Neutral  30-May-10    Won  
2     Sawai Mansingh Stadium, Jaipur     Home  16-Oct-13    Won  
3  M. Chinnaswamy Stadium, Bangalore     Home  02-Nov-13    Won  
4              Eden Gardens, Kolkata     Home  06-Nov-13    Won

In [14]:
df.shape

(43, 9)

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43 entries, 0 to 42
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   No.       43 non-null     int64 
 1   Score     43 non-null     object
 2   Against   43 non-null     object
 3   Position  43 non-null     int64 
 4   Innings   43 non-null     int64 
 5   Venue     43 non-null     object
 6   Ground    43 non-null     object
 7   Date      43 non-null     object
 8   Result    43 non-null     object
dtypes: int64(3), object(6)
memory usage: 3.1+ KB


In [43]:
df['Score']=df['Score'].str.replace('*','')



C:\Users\Nitij\AppData\Local\Temp\ipykernel_2792\3827732414.py:1: FutureWarning:

The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.



In [49]:
df['Score']=df['Score'].astype(str).astype(int)     

In [50]:
df['Score']

0     114
1     101
2     141
3     209
4     177
5     111
6     264
7     138
8     137
9     106
10    150
11    171
12    124
13    123
14    124
15    104
16    125
17    147
18    102
19    208
20    118
21    115
22    100
23    137
24    111
25    152
26    162
27    111
28    133
29    122
30    140
31    102
32    104
33    103
34    176
35    127
36    212
37    159
38    119
39    161
40    127
41    101
42    120
Name: Score, dtype: int32

In [51]:
import dtale
dtale.show(df)

C:\Users\Nitij\anaconda3\lib\site-packages\dtale\charts\utils.py:644: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.

C:\Users\Nitij\anaconda3\lib\site-packages\dtale\charts\utils.py:620: FutureWarning:

Not prepending group keys to the result index of transform-like apply. In the future, the group keys will be included in the index, regardless of whether the applied function returns a like-indexed object.
To preserve the previous behavior, use

	>>> .groupby(..., group_keys=False)

To adopt the future behavior and silence this warning, use 

	>>> .groupby(..., group_keys=True)

C:\Users\Nitij\anaconda3\lib\site-packages\dtale\charts\utils.py:620: FutureWarning:

Not prepending group keys to the result index of transform-like apply. In the future, the group keys will be included in the index, regar

# OBSERVATIONS

1) Highest run is 264 against Srilanka.

2) He scored 4 time more than 200 runs, all double centuries came under home ground.

3) Out of 43 innings he usually open the innings but few time he had batted at 4th position and 6th position as well.

4) He score on an average 136.9 runs per innings.

5) He like to score hundred against Austrailia and Srilanka.

6) He scored centuries more in India i.e Home ground compared to against and neutral but Edgbaston Cricket Ground, Birmingham seems to luckier and favourite for him as he had cored 3 100+ score there.

7) Out 0f 43 centuries innings he lost only 8 times , India won 34 times whenever Rohit Sharma scores a century.